### PR TITLE
feat(batch): fargate support for jobs

### PR DESCRIPTION
**Pull request by kokachev**
Added Fargate support for Batch jobs.

closes: #13590

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by kokachev** [comment]
@iliapolo review?

**Comment by iliapolo** [comment]
> @iliapolo review?

Apologies for the delay. Thanks for creating this PR! 

I should be able to get a first round in soon 👍

**Comment by kokachev** [comment]
> > @iliapolo review?
> 
> @kokachev Apologies for the delay. Thanks for creating this PR!
> 
> I should be able to get a first round in soon 👍

@iliapolo can someone else review it?

**Comment by kokachev** [comment]
@iliapolo should I close this PR?

**Comment by iliapolo** [comment]
@kokachev No - I still have this on my radar. Would be good if you resolved the conflicts that arose in the meantime though :)

File: packages/@aws-cdk/aws-batch/README.md [hunk 1]
```diff
@@ -74,6 +74,21 @@ const spotEnvironment = new batch.ComputeEnvironment(stack, 'MySpotEnvironment',
...
 ```
 
+### Fargate Compute Environment
+
+It is possible to have AWS Batch submit jobs to be run on Fargate compute resources. Below is an example of how this can be done:
+
+```ts
+const vpc = new ec2.Vpc(this, 'VPC');
+
+const spotEnvironment = new batch.ComputeEnvironment(stack, 'MyFargateEnvironment', {
```
Comment (iliapolo) [inline packages/@aws-cdk/aws-batch/README.md:84]:
```suggestion
const fargateSpotEnvironment = new batch.ComputeEnvironment(stack, 'MyFargateEnvironment', {
```

File: packages/@aws-cdk/aws-batch/lib/compute-environment.ts [hunk 2]
```diff
@@ -18,6 +18,16 @@ export enum ComputeResourceType {
...
+
+  /**
+   * Resources will be Fargate resources.
+   */
+  FARGATE = 'FARGATE',
+
+  /**
+   * Resources will be Fargate resources.
+   */
+  FARGATE_SPOT = 'FARGATE_SPOT',
```
Comment (iliapolo) [inline packages/@aws-cdk/aws-batch/lib/compute-environment.ts:30]: What does *spot* mean in the context of *fargate*? would be nice to have a sentence or two about it.

File: packages/@aws-cdk/aws-batch/test/compute-environment.test.ts [hunk 3]
```diff
@@ -82,6 +82,164 @@ describe('Batch Compute Evironment', () => {
       });
     });
   });
+  describe('using fargate resources', () => {
```
Comment (iliapolo) [inline packages/@aws-cdk/aws-batch/test/compute-environment.test.ts:85]: We also need an integ test. Add a compute environment to https://github.com/aws/aws-cdk/blob/master/packages/@aws-cdk/aws-batch/test/integ.batch.ts#L30

**Review by iliapolo — CHANGES_REQUESTED** [review]
Nice. Very clean and straight forward 👍

**Comment by kokachev** [comment]
will update the PR.

**Comment by brendonparker** [comment]
Looking forward to being able to play with this :)

File: packages/@aws-cdk/aws-batch/lib/job-definition.ts [hunk 4]
```diff
@@ -458,6 +523,9 @@ export class JobDefinition extends Resource implements IJobDefinition {
       user: container.user,
       vcpus: container.vcpus || 1,
       volumes: container.volumes,
+      fargatePlatformConfiguration: container.fargatePlatformConfiguration ? {
```
Comment (brendonparker) [inline packages/@aws-cdk/aws-batch/lib/job-definition.ts:526]: Sorry, my github-fu is weak. Not sure how to put a comment on code that isn't a part of the PR.

But above, when setting the `resourceRequirements`, don't we need a way to pass those in for fargate?
```
resourceRequirements: container.gpuCount
  ? [{ type: 'GPU', value: String(container.gpuCount) }]
  : undefined
```

Also, pretty sure that `vcpus` needs to be not set

**Comment by robertd** [comment]
Looking forward to this PR being merged in 👍 ... just in time for what I'm working on. 😄

**Comment by DDynamic** [comment]
@iliapolo @brendonparker since it has been a little while since the author has responded, I've opened PR #15848 that responds to all of the feedback left of this MR in hopes that this will be brought to completion soon. A review as soon as you get a chance would be greatly appreciated.

**Comment by robertd** [comment]
@DDynamic sweet 👍😊

**Comment by iliapolo** [comment]
Closing in favor of https://github.com/aws/aws-cdk/pull/15848

**Pull request closed without merging**